### ISSUE TITLE
extremely imbalanced problems processing

### ISSUE THREAD

**Issue by licsh** [automl/auto-sklearn#726]
auto-sklearn can handle the problems of extremely imbalanced dataset like 1:10000  in classification? I have met the practical problems recently, so I want to know how auto-sklearn solve this problem

**Comment by Loopiezlol** [comment]
never used `auto-sklearn`, just lurking around. 
I think you have some control with regards to this matter through two parameters:
- metric used: while the default one is _accuracy_, you might wanna take a look at _f1 score_ for imbalanced classification
- resampling strategy: as per https://github.com/automl/auto-sklearn/issues/350#issuecomment-337888282, I think using the `cv` (cross-validation) option will ensure stratification. what this means is that every split of data used for training will have the same class ratio as your original dataset

take a look at this API to see what other parameters you can mess around with: https://automl.github.io/auto-sklearn/master/api.html#classification

**Comment by mfeurer** [comment]
Yes, Auto-sklearn should be able to handle (i.e. run and not crash) such problems. Some further comments:

* Auto-sklearn by default configures whether the data will be re-weighted
* You can choose a metric which is more appropriate for imbalanced data than regular accuracy. Other possibilities than the f1 score are area under curve or balanced accuracy
* Yes, Auto-sklearn uses stratified sampling. This also means that you need to have more positive samples than splits.
* Auto-sklearn cannot use any further preprocessing algorithms like SMOTE because the scikit-learn pipeline on which Auto-sklearn is buit does not allow changing the number of samples.

Thanks @Loopiezlol for answering the question! Please reopen @licsh if you have any further questions.

**Comment by UnixJunkie** [comment]
Is it possible to give a list of weights that should be tried for a given class?
I have some data where very heavy reweighting of the under-represented class is necessary to get any good classifier.

**Comment by mfeurer** [comment]
Hey @UnixJunkie, could you please open a new issue? We do not follow up or answer closed issues.
